### DISCUSSION TITLE
See table information: constraints, indexes, etc

### DISCUSSION THREAD

**Discussion by JPFrancoia** [tconbeer/harlequin#586] [Q&A, Troubleshooting, Support]
Hi!

I started using Harlequin and it's a pretty nice tool. I feel it's missing a feature though, but maybe I missed that part in the docs.
Is it possible to view the schema of a table, with information about constraints and indexes (and possibly size of the data)?

With psql if you run the shortcut `\d my_table` you get the schema of the table + information about the indexes and constraints of the table. Something like this:

![screen-2024-06-30-15-47-38](https://github.com/tconbeer/harlequin/assets/2399090/783d8204-4b87-4357-94d4-2d4785b9f59a)

If you run `\d+` you also get the size of all the tables in the current schema. This would be really useful, otherwise we'll still need tools like psql or DBeaver to get this kind of information.

**Comment by tconbeer** [comment]
I just released harlequin-postgres v1.2.0, which includes new catalog interactions that provide the same info as psql's `\d+`. Here's a demo:
https://www.tella.tv/video/postgres-adapter-catalog-interactions-for-d-1-hpyx
